vhdbuilder/packer/test/linux-vhd-content-test.sh

```diff
@@ -296,6 +296,39 @@ testFips() {
   echo "$test:Finish"
 }
 
+testCloudInit() {
+  test="testCloudInit"
+  echo "$test:Start"
+  os_sku=$1
+
+  # Limit this test only to Mariner or Azurelinux 
+  if [[ "$os_sku" == "CBLMariner" || "$os_sku" == "AzureLinux" ]]; then
+    FILE=/var/log/cloud-init.log
+    if test -f "$FILE"; then
+      echo "Check cloud-init log exist."
+      if grep 'WARNING\|ERROR' $FILE; then
+        msg=$(grep 'WARNING\|ERROR' $FILE)
+        err $test "Cloud-init log has WARNING/ERROR: ${msg}."
+      else
+        echo "Cloud-init log is OK."
+      fi
+    else
+      err $test "Check cloud-init log does not exist."
+    fi
+
+    if cloud-init status --wait; then
+      echo "Cloud-init status is OK."
+    else
+      status=$(cloud-init status --wait)
+      err $test "Cloud-init status error with ${status}."
+    fi
```

Comment:
Same as above: I don't like using the same command twice. (Again; untested code).
```suggestion
    local cloud_init_output=$(cloud-init status --wait)
    local cloud_init_status=$?
    if [ ${cloud_init_status} -eq 0 ]; then
      echo "Cloud-init status is OK."
    else
      err $test "Cloud-init status error with ${cloud_init_status}."
    fi
```